### PR TITLE
Add basic struct generation

### PR DESCRIPTION
**Pull request by marv7000**
# What does this PR change?

This PR adds a basic Rust struct generation by parsing a YAML schema using Serde and then reformatting the output.

Tick the applicable box:
- [x] Add new feature
- [ ] Security changes
- [ ] Tests
- [ ] Documentation changed
- [ ] General Maintenance

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -24,10 +25,14 @@ fn main() {
 
     // Welcome Message
     println!(
-        "{} \n(c) The Nezara Project. (github.com/The-Nezara/Project)\n
```
Comment (ByteOtter) [inline src/main.rs:27]: I knew I had a typo in there somewhere!

**Review by ByteOtter — CHANGES_REQUESTED** [review]
So I have looked over it and found that it panics on my machine.
Here's the shortened backtrace, maybe you can figure this one out:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/bindgen.rs:94:70
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: core::result::unwrap_failed
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/result.rs:1651:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/result.rs:1076:23
   4: thanix::bindgen::gen
             at ./src/bindgen.rs:94:26
   5: thanix::main
             at ./src/main.rs:35:23
   6: core::ops::function::FnOnce::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Also I would like you to format the code using `cargo fmt` if not already done so. The workflow shows errors so if you could check that I would greatly appreciate this.

Apart from that, great work and thanks for the help!

**Comment by marv7000** [comment]
I think you just need to create the "output/" directory, that's an easy fix though.
I did run `cargo fmt` over it, but clippy will complain because most of the struct fields are still not marked with `#[allow(dead_code)]`.

**Comment by ByteOtter** [comment]
> I think you just need to create the "output/" directory, that's an easy fix though. I did run `cargo fmt` over it, but clippy will complain because most of the struct fields are still not marked with `#[allow(dead_code)]`.

You were right that did the trick. Stupid that I didn't think of that. 🤦🏼 

Clippy will complain about dead code but as these are classified as warnings they can be ignored for now. But I did see real errors in the workflow output so maybe check tomorrow.

**Comment by marv7000** [comment]
Should work now, seems like `File::write` is not optimal.

**Comment by ByteOtter** [comment]
Great work on fixing that this quickly. 
As this is a complete feature now, I would suggest we merge it tomorrow.
Would you say that adding some tests for this feature would be even possible? Not today of course ...

**Comment by marv7000** [comment]
Sure, I'll get to it.

**Review by ByteOtter — APPROVED** [review]

**Review by ByteOtter — APPROVED** [review]
LGTM. We're gonna rework the splitting  of these structs into different modules later.

**Pull request merged**